src/json-api/json-api-module.ts

```diff
@@ -297,9 +301,28 @@ export class JsonApiModule {
       latestCommit,
       latestSuccessfullyDeployedCommit,
       latestActivityTimestamp: branch.latestActivityTimestamp,
+      token: this.tokenGenerator.branchToken(project.id, branch.name),
     };
   }
 
+  public async getLatestDeploymentIdForBranch(projectId: number, branchName: string) {
+    const branch = await this.projectModule.getBranch(projectId, branchName);
+    if (!branch) {
+      throw new Error(`Unable to find branch ${branchName} for project ${projectId}`);
+    }
+    const deployment = await this.deploymentModule.getLatestSuccessfulBranchDeployment(
+      projectId,
+      branchName,
+      branch.latestCommit.id,
```

Comment:
This is a bit awkward, but:

It will always find the latest succesful deployment, if it has been done for a push with `branchName`. See the comments in the implementation for `getLatestSuccessfulBranchDeployment`.